### ISSUE TITLE
In a property setter $self binding points to the setter

### ISSUE THREAD

**Issue by vszekeress-sportradar** [AvaloniaUI/Avalonia#11872]
Reopening [#7809](https://github.com/AvaloniaUI/Avalonia/issues/7809), following code now causes compile time error:

```
<Styles xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        x:CompileBindings="True">
    <Style Selector=":is(ProgressBar).SlimStyle">
        <Setter Property="CornerRadius" Value="{Binding $self.Height}" />
...
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/88656333/0d3b9408-8139-4c18-b91a-e0ae59587021)

**Comment by timunie** [comment]
Which Avalonia version?

**Comment by timunie** [comment]
In general, please fill out the templates we provide. They are there for a reason. Thx 🙏

**Comment by timunie** [comment]
Works for me using 11.0.

Note: if you are using Avalonia 0.10, this will not be fixed there. So I'm closing this as I have no more info about your usecase.

**Comment by vszekeress-sportradar** [comment]
I confirm that the issue is fixed in 11.0, thanks.
